### ISSUE TITLE
1.15.17 scrollable.dart is crashing

### ISSUE THREAD

**Issue by blaneyneil** [flutter/flutter#53058]
i can't replicate this on my dev builds for both iOS and Android.  It's only happening on production builds for iOS.  luckily i just got my error report since i couldn't log to the console.  two errors:

```
_AssertionError
Failed assertion: boolean expression must not be null
scrollable.dart in ScrollableState.didUpdateWidget at line 402
framework.dart in StatefulElement.update at line 4690
framework.dart in Element.updateChild at line 3201
framework.dart in ComponentElement.performRebuild at line 4527
```

```
_CastError
type 'ScrollableState' is not a subtype of type '_FixedExtentScrollableState' in type cast
list_wheel_scroll_view.dart in _FixedExtentScrollPosition._getItemExtentFromScrollContext at line 389
list_wheel_scroll_view.dart in new _FixedExtentScrollPosition at line 382
list_wheel_scroll_view.dart in FixedExtentScrollController.createScrollPosition at line 289
scrollable.dart in ScrollableState._updatePosition at line 366
scrollable.dart in ScrollableState.didChangeDependencies at line 382
framework.dart in StatefulElement._firstBuild at line 4661
```

```
flutter doctor -v

[✓] Flutter (Channel beta, v1.15.17, on Mac OS X 10.15.3 19D76, locale en-US)
    • Flutter version 1.15.17 at /Users/neil/flutter
    • Framework revision 2294d75bfa (2 weeks ago), 2020-03-07 00:28:38 +0900
    • Engine revision 5aff311948
    • Dart version 2.8.0 (build 2.8.0-dev.12.0 9983424a3c)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/neil/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C504
    • CocoaPods version 1.8.3

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 44.0.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
```

**Comment by Piinks** [comment]
Hi @blaneyneil thanks for reporting.
Can you provide a minimal, runnable code sample? It will be very hard to tell why this is happening without a reproduction. Thanks!

**Comment by blaneyneil** [comment]
I never called it directly, so i'm not exactly sure where it was called from.  the offending screen had listviews and date and time pickers, and had never had a problem before 1.15.17.  i wish i could offer more details, but i couldn't replicate in my local builds so i couldn't output anything to the console.  the output above was from Sentry.  

but once it was on production, on an actual iOS device, the offending page crashed for everyone.

**Comment by Piinks** [comment]
Thanks for the feedback @blaneyneil 

FYI @xster

**Comment by goderbauer** [comment]
@blaneyneil What do you mean by this only happens in production? Production builds usually have assertions disabled so you really shouldn't be seeing any assertion errors there.

**Comment by blaneyneil** [comment]
I couldn't ever replicate the crash when using my simulator, using a Debug build.  Only when it was a Release build (iOS only) on an actual device did it trigger the crash.  It's possible the Sentry _AssertionError came from one of our debuggers who was using our dev Release build.

**Comment by Piinks** [comment]
@blaneyneil is it possible to provide a runnable snippet of code that is crashing in production on iOS? It looks like it should contain a `ListWheelScrollView`. I haven't been able to reproduce this yet.

**Comment by blaneyneil** [comment]
I didn't have any direct calls to `ListWheelScrollView`.   But the page that was crashing for me luckily had a near clone, and i can't get that clone to crash.  The only significant  difference between the two was that the crashing page declared two `ListView` Widgets (which would account for the `scrollable.dart` crash) in the `StatefulWidget` instead of one and then called them in the body like so:

```
body: Container(
          color: Colors.white,
          child: Stack(
            children: <Widget>[

              _edit ? _form1() : _form2(),

```

where both _form1 and _form2 are:

```
return ListView(
      controller: _scrollController,
      children: <Widget>[
```

nothing special about anything, except that i only initiated a single:

`final ScrollController _scrollController = ScrollController();
`
and attached that same one to both ListViews.  That's really the only thing i can think of that might be the root cause.  I've since declared separate ScrollControllers for each ListView, but because my production build was breaking i had to scramble, and so reverted to the Stable build from Beta.  I'm going to try and go back to Beta tomorrow.

**Comment by goderbauer** [comment]
@blaneyneil How did the switch to Beta go? Any luck in reproducing this?

**Comment by blaneyneil** [comment]
Can't until tomorrow, fyi.  In the middle of stuff i have to deploy first.

**Comment by blaneyneil** [comment]
just tested an iOS Release build, after switching to the beta channel, and no, the page in question no longer crashes.  however, beta is now 1.17.0.  and as mentioned above, i've since initialized and disposed of  a `ScrollController` for each `ListView`.   i can't risk reverting to a single 'ScrollController' as a control variable though, since this is a live app.  so not sure whether it was another ScrollController or 1.17.0 that fixed it.

**Comment by blaneyneil** [comment]
@goderbauer pinging you since i'd already triggered the no-response bot.

**Comment by Piinks** [comment]
> just tested an iOS Release build, after switching to the beta channel, and no, the page in question no longer crashes. however, beta is now 1.17.0. and as mentioned above, i've since initialized and disposed of a `ScrollController` for each `ListView`. i can't risk reverting to a single 'ScrollController' as a control variable though, since this is a live app. so not sure whether it was another ScrollController or 1.17.0 that fixed it.

Hey @blaneyneil it sounds like this is no longer happening based on your comment. Am I understanding that correctly?

**Comment by blaneyneil** [comment]
Correct.  I just can't tell whether the fix was something in 1.17.0 or the separate ScrollControllers.

**Comment by Piinks** [comment]
> Correct. I just can't tell whether the fix was something in 1.17.0 or the separate ScrollControllers.

Thanks for following up! I am glad that the crash is no longer happening. In that case, I don't know that this bug is actionable currently. I am going to close it for now but if you experience this crash again, let us know and we will reopen it!
